language-plutus-core/src/Language/PlutusCore/TypeCheck/Internal.hs

```diff
@@ -421,5 +428,6 @@ checkTypeM
 -- ---------------------------------------------
 -- [check| G !- term : vTy]
 checkTypeM ann term vTy = do
+    checkKindM ann (ann <$ unNormalized vTy) $ Type ()
```

Comment:
Seems like a reasonable check, but it should probably go in the rule preconditions in the comment too.

Reply:
I just forgot to remove it. The point is that we do not check that normalization preserves the kind of an input type, because it would be an internal error if normalization did not preserve the kind and we do not check for those. So I was wondering if we should make such additional checks. We then need them in other places too.

Reply:
But if we decide to add such checks, then let's do that in a different PR. Here I'll just remove the check.

Reply:
Removed the check.